aerogear/mobile-security-service-operator
# Make watch/cache just check specific namespaces

## Motivation
- https://issues.jboss.org/browse/AEROGEAR-9089
- **IMPORTANT:** Blocked by `https://github.com/operator-framework/operator-sdk/pull/1388`
## What
Ensure that the operator will just watch the specific namespaces

## Why
Attend requirement

## How
* upgrade the lib https://github.com/kubernetes-sigs/controller-runtime to the version v0.2.0-alpha.0 which has the impl https://github.com/kubernetes-sigs/controller-runtime/commit/fc804a411c06919561edbcb0453601c436556613 
* use the new API to the cache which allows us to do it
* upgrade to use the master branch of operator-sdk dep and fix break changes 

## Verification Steps
* Check the README
* Use this PR and build a dev image to test it
* Update the operator.yaml file to use this new image
* Install the operator by running "make create-all"
* Check that all were installed with success
* Create the default app namespace "oc new-project mobile-security-service-apps"
* Run make create-app
* Check that the configMap was created in this namespace for this app
* Create another namespace with any other name
* Run `make create-all` inside of this new namespace
* Ensure that the configMap was NOT created. 

## Checklist:


## Progress


## Additional Notes
